controllers/cloud.redhat.com/clowdjobinvocation_controller.go

```diff
@@ -142,15 +142,17 @@ func (r *ClowdJobInvocationReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	// Determine if the ClowdApp containing the Job is ready
 	if !app.IsReady() {
-		r.Recorder.Eventf(&app, "Warning", "ClowdAppNotReady", "ClowdApp [%s] is not ready", cji.Spec.AppName)
-		r.Log.Info("App not yet ready, requeue", "jobinvocation", cji.Spec.AppName, "namespace", app.Namespace)
-		readyErr := errors.NewClowderError(fmt.Sprintf("The %s app must be ready for CJI to start", cji.Spec.AppName))
-		if condErr := SetClowdJobInvocationConditions(ctx, r.Client, &cji, crd.ReconciliationFailed, readyErr); condErr != nil {
-			return ctrl.Result{}, condErr
-		}
+		if !cji.Spec.RunOnNotReady {
```

Comment:
The negative conditional nested in a negative conditional is hard to read and reason about. And this makes a large method larger. I think you could improve this code by factoring this logic out into a method (reduce method complexity) and recast the conditionals as positive non-nested guards. Here's a sketch (didn't test this):

Recast as a method
```golang
func (r *ClowdJobInvocationReconciler) HandleNotReady(... fill out correct args here ...) {
	if app.IsReady() {
		return nil
	}
	if cji.Spec.RunOnNotReady {
		return nil
	}
	r.Recorder.Eventf(&app, "Warning", "ClowdAppNotReady", "ClowdApp [%s] is not ready", cji.Spec.AppName)
	r.Log.Info("App not yet ready, requeue", "jobinvocation", cji.Spec.AppName, "namespace", app.Namespace)
	readyErr := errors.NewClowderError(fmt.Sprintf("The %s app must be ready for CJI to start", cji.Spec.AppName))
	if condErr := SetClowdJobInvocationConditions(ctx, r.Client, &cji, crd.ReconciliationFailed, readyErr); condErr != nil {
		return condErr
	}

	// requeue with a buffer to let the app come up
	return readyErr
}
```

And calling it within your main method
```golang
notReadyError := HandleNotReady(... fill out correct args here ...)
if notReadyError != nil {
	return ctrl.Result{Requeue: true}, notReadyError
}
```suggestion
		if !cji.Spec.RunOnNotReady {
```
```